### PR TITLE
Add undo/redo buttons to the top bar

### LINKED ISSUE #11222: Add Undo/Redo buttons
Add Undo/Redo buttons to the graph editor

image

They should go on the top bar to the right of Refresh/Play.

As part of this story can we re-size refresh/play width so the curve of the inner circle matches the edge?

image

### PR DESCRIPTION
**Pull request by vitvakatu**
### Pull Request Description

Closes #11222 

https://github.com/user-attachments/assets/66f8dfd1-a7c4-497a-8ff6-a22f56dbb1dd

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] Unit tests have been written where possible.
- [x] If meaningful changes were made to logic or tests affecting Enso Cloud integration in the libraries, 
      or the Snowflake database integration, a run of the [Extra Tests](https://github.com/enso-org/enso/actions/workflows/extra-nightly-tests.yml) has been scheduled.
  - If applicable, it is suggested to paste a link to a successful run of the Extra Tests.

### CODE REVIEW

File: app/gui/src/project-view/components/UndoRedoButtons.vue [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      />
+    </template>
+    <template #right>
+      <SvgButton
+        title="Write All"
+        class="iconButton"
+        name="redo"
+        draggable="false"
+        :disabled="!graphStore.undoManager.canRedo.value"
+        @click.stop="graphStore.undoManager.redo()"
```
Comment (farmaazon) [inline app/gui/src/project-view/components/UndoRedoButtons.vue]: nit: undo call is without `()` but redo is with.

File: app/gui/src/project-view/components/UndoRedoButtons.vue [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        class="iconButton"
+        name="undo"
+        draggable="false"
+        :disabled="!graphStore.undoManager.canUndo.value"
+        @click.stop="graphStore.undoManager.undo"
+      />
+    </template>
+    <template #right>
+      <SvgButton
+        title="Write All"
```
Comment (farmaazon) [inline app/gui/src/project-view/components/UndoRedoButtons.vue]: Wrong title.

File: app/gui/src/project-view/util/data/observable.ts [hunk 3]
```diff
@@ -2,7 +2,18 @@
...
 export type Events<O extends ObservableV2<any>> = O extends ObservableV2<infer E> ? E : never
+/** A list of all possible events of the observable, in fixed (but unspecified) order. */
+export type ListOfEvents<O extends ObservableV2<any>> = TupleFromUnion<keyof Events<O>>
+
+type TupleFromUnion<T, L = LastOf<T>, N = [T] extends [never] ? true : false> =
+  true extends N ? [] : [...TupleFromUnion<Exclude<T, L>>, L]
+type UnionToIntersection<U> =
+  (U extends any ? (k: U) => void : never) extends (k: infer I) => void ? I : never
+type LastOf<T> =
+  UnionToIntersection<T extends any ? (x: T) => void : never> extends (x: infer L) => void ? L
```
Comment (farmaazon) [inline app/gui/src/project-view/util/data/observable.ts]: I admit I don't understand this. Why this `UnionToIntersection` takes a function type, not a union? Any chance of explanation?

**Review by farmaazon — APPROVED** [review]

File: app/gui/src/project-view/util/data/observable.ts (hunk 3 above)
Comment (vitvakatu) [reply app/gui/src/project-view/util/data/observable.ts]: Nope, not a chance.

After researching it a bit, I think the current implementation is too fragile to use, I will come up with something else.

**Pull request merged**